### PR TITLE
Lab 1 luliangLi

### PR DESCRIPTION
**Pull request by luliangLi**
Fixes # 24

Review of colleague's PR #152 

#### Changes proposed in this PR:
- Completed lab 1 for both fib and negaFib
- Unit test cases

### CODE REVIEW

**Review by juliaogris — CHANGES_REQUESTED** [review]
Looks pretty good.
Great job including table driven tests and so many test cases!

Please read [collected lab1 review comments](https://github.com/anz-bank/go-course/wiki/Lab-1:-Collected-Code-Review-Comments):
- Fix PR description (add issue number and link to review of colleague's PR)

Your solution could be simplified with:
```go
	sign := // 1 for positive n ; -1 for negative n
	n1, n2 := 0, 1
	for j := 0; j < n*sign; j++ {
		n1, n2 = n2, n1+sign*n2
		fmt.Fprintln(out, n1)
	}

```

File: 01_fib/luliangLi/main_test.go
Comment (eastrd) [inline 01_fib/luliangLi/main_test.go:28]: Not needed as a variable assigns to itself.

**Review by eastrd — CHANGES_REQUESTED** [review]
Line 28 on `main_test.go` looks extra, otherwise LGTM.

File: 01_fib/luliangLi/main_test.go
Comment (luliangLi) [reply 01_fib/luliangLi/main_test.go:28]: Thanks for reviewing my code. 
Not sure if you encountered the issue or not if removing the tricky "test := test", while I got the following issue anyhow:
Using the variable on range scope `test` in function literal (scopelint)
			fib(test.input)

So I googled it and found the solution in the link: https://github.com/kyoh86/scopelint., which taught me to add the line to pin an external variable to an internal one.

Comment (eastrd) [reply 01_fib/luliangLi/main_test.go:28]: Sorry LuliangLi, my bad here.

**Review by eastrd — DISMISSED** [review]
Looks good to me

**Comment by anzboi** [comment]
I'm a little unsure about the state of this PR. It looks like @eastrd approved your PR, then you added the `Merge me` label yourself, this is just for instructors to tell you to merge it when you can. @eastrd also approved the PR, this should be left to instructors. I've removed `Merge me` for now and dismissed the approval. Wait for @juliaogris to approve and add `Merge me` again.

**Review by juliaogris — APPROVED** [review]
The lab is solved.
I've fixed up your PR description (still had some formatting issues).
Also, I'm a little confused by the recursive solution and the nested ifs in `fib`.
To me a single function similar to the listing below would have been clearer - but I understand that this is to an extend a question of taste, so no further updates required:
```go
func fib(n int) {
        if n == 0  {
		return
	}
          
	sign := 1
	if n < 0 {
		sign = -1
	}

	n1, n2 := 0, 1
	for j := 0; j < n*sign; j++ {
		n1, n2 = n2, n1+sign*n2
		fmt.Fprintln(out, n1)
	}
}
```

**Comment by juliaogris** [comment]
The go-course is now closed. Thank you very much for participating.

**Pull request closed without merging**
